### DISCUSSION TITLE
how to use slugs

### DISCUSSION THREAD

**Discussion by AbhimanyuAryan** [nuxt/content#2557] [Q&A] [answered]
I have an empty `pages/deeplearning/_slug.vue`. I don't know what to put inside it. The documentation is not very helpful

I have my blog posts in `content/deeplearning/post1.md` and `content/deeplearning/post2.md`

**pages/deeplearning.vue**
```html
<template>
  <main>
    <ContentList path="/deeplearning" v-slot="{ list }">
      <div class="cards">
        <nuxt-link
          v-for="item in list"
          :key="item._id"
          :to="`/deeplearning/${item._path?.split('/').pop() || 'fallback-path'}`"
          class="card-link"
        >
          <div class="card">
            <h2>{{ item.title }}</h2>
            <p>{{ item.description }}</p>
            <p>Author: {{ item.author }}, Date: {{ new Date(item.date).toLocaleDateString() }}</p>
          </div>
        </nuxt-link>
      </div>
    </ContentList>
  </main>
</template>
```

The goal is to show markdown content on /post1 route and /post2 route

![slug](https://github.com/nuxt/content/assets/8083613/193a45d9-aa04-4a61-a011-5e50d3b5a8bc)

here's my blog: https://github.com/AbhimanyuAryan/abhimanyuaryan.github.io

**Comment by TheAlexLichter** [comment]
You would define your file as `[slug].vue` in Nuxt 3. Take a look [at the routing docs](https://nuxt.com/docs/getting-started/routing) for more info.

**Comment by farnabaz** [comment]
`_slug` is Nuxt 2 style, You need to define your page in Nuxt 3 way `[slug].vue`.
Inside `[slug].vue` you can sue `<ContentDoc />` to render the content:

```vue
<template>
  <main>
    <ContentDoc />
  </main>
</template>
```

`ContentDoc` uses `$route.path` to find and render the content. [See](https://content.nuxt.com/usage/render)

**Comment by AbhimanyuAryan** [comment]
Hi @farnabaz @manniL thanks for the reply. I used `[slug].vue` but it doesn't show content of post1 when I click on the card. As you can see the gif

**Comment by farnabaz** [comment, marked as answer]
Try moving `deeplearning.vue` to `deeplearning/index.vue`

**Comment by AbhimanyuAryan** [comment]
Thanks @farnabaz 🙏🏻 it worked 🎉
